hedgehog/src/Hedgehog/Internal/Tree.hs

```diff
@@ -301,11 +301,30 @@ splits = \case
     ([], x, xs) :
     fmap (\(as, b, cs) -> (x : as, b, cs)) (splits xs)
 
-dropOne :: Monad m => [NodeT m a] -> [TreeT m [a]]
-dropOne ts = do
-  (xs, _y, zs) <- splits ts
-  pure . TreeT . pure $
-    interleave (xs ++ zs)
+-- | @removes n@ computes all ways we can remove chunks of size @n@ from a list
```

Comment:
I get what this is doing, but not quite sure it's clear from this sentence!

'all the ways we can remove chunks of size n' sounds like you'd have:
```
-- > removes 2 [1..4] = [[3,4],[1,4],[1,2]]
```

Reply:
Fair. Suggested rewording?

Reply:
Would "Splits a list into chunks of size `n` and returns all possible lists where one of these chunks has been removed" be less confusing?

Reply:
Again, I don't mind the behavior, but I can't reconcile how chunks of size `2` are removed from:
```-- > removes 2 [1..5] == [[3,4,5],[1,2,5],[1,2,3,4]]```

Ah ok, that is this part yes?
> Note that the last chunk we delete might have fewer elements than @n@.

Ok, I think I get it now, posting this ramble for posterity

Reply:
Yeah, it removed the chunk `5 : nothingMoreToRemove` :)